packages/netlify-cms-core/src/components/Collection/FilterControl.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import { translate } from 'react-polyglot';
+import {
+  buttons,
+  Dropdown,
+  DropdownCheckedItem,
+  StyledDropdownButton,
+} from 'netlify-cms-ui-default';
+
+const FilterButton = styled(StyledDropdownButton)`
+  ${buttons.button};
+  ${buttons.medium};
+  ${buttons.grayText};
+  font-size: 14px;
+
+  &:after {
+    top: 11px;
+  }
+`;
+
+const FilterControl = ({ viewFilters, t, onFilterClick, filter }) => {
+  return (
+    <Dropdown
+      renderButton={() => {
+        return <FilterButton>{t('collection.collectionTop.filter')}</FilterButton>;
+      }}
+      closeOnSelection={false}
+      dropdownTopOverlap="30px"
+      dropdownPosition="left"
+    >
+      {viewFilters.map(viewFilter => {
+        return (
+          <DropdownCheckedItem
+            key={viewFilter.field}
+            label={viewFilter.label}
+            id={viewFilter.field}
+            checked={filter.get(viewFilter.field, false)}
+            onClick={() => onFilterClick(viewFilter.field)}
```

Comment:
@erezrokah I added sample filters. I just noticed that we are not passing `viewFilter.pattern` while calling` onFilterClick`. As a result `filter` prop in redux doesn't have pattern.  Shouldn't collections be filtered upon pattern?
Current filter state in redux:
![filter](https://user-images.githubusercontent.com/10570968/81934561-ca896f80-960c-11ea-9a2d-171fc877108c.png)

Reply:
Yeah I don't fully see how @erezrokah envisioned this. 
The `onFilterClick` should imho take the full `viewFilter` object. This needs to be passed down in the `filterByField` entry action. As a result the redux store needs to be modified since I don't think the current `Filter` definition would work. We would need a `FilterObject` very similar to how the `SortObject` is defined but with an additional pattern property. From there on out it should be smooth sailing.

Reply:
You're both correct - I failed to finalise that part...

Reply:
Gotcha! I assumed that would be the case.

Reply:
@erezrokah could you please make the required changes and I'll be able to work on the frontend part from there? Thank you

Reply:
Sure thing @BA1RY, I think I'll be able to do it tomorrow